### PR TITLE
protobuf: add version 3.21.12

### PR DESCRIPTION
**Pull request by gjasny**
Specify library name and version:  **protobuf/3.21.12**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/protobuf/all/conandata.yml [hunk 1]
```diff
@@ -21,6 +24,11 @@ sources:
     url: "https://github.com/protocolbuffers/protobuf/archive/refs/tags/v3.17.1.tar.gz"
     sha256: "036d66d6eec216160dd898cfb162e9d82c1904627642667cc32b104d407bb411"
 patches:
+  "3.21.12":
+    - patch_file: "patches/protobuf-3.21.12-upstream-macos-macros.patch"
+      patch_description: "Handle case where macOS SDK macros may conflict with protobuf message types"
+      patch_type: "bugfix"
+      patch_source: "https://github.com/protocolbuffers/protobuf/pull/10103"
```
Comment (memsharded) [inline recipes/protobuf/all/conandata.yml:31]: I don't see the connection of the patch with https://github.com/protocolbuffers/protobuf/pull/10103, that PR was merged a year ago, are you sure this is the correct link? Could you please elaborate a bit more?

**Review by memsharded — COMMENTED** [review]
Thanks for your contribution

File: recipes/protobuf/all/conandata.yml (hunk 1 above)
Comment (gjasny) [reply recipes/protobuf/all/conandata.yml:31]: I did not write the patch, I only copied it like everyone else before me did. Although committed to `master` the patch is not part of the `21.x` branch of protobuf, yet: [code](https://github.com/protocolbuffers/protobuf/blob/21.x/src/google/protobuf/port_def.inc#L870-L873)

Comment (mayeut) [reply recipes/protobuf/all/conandata.yml:31]: c.f. #16453

**Review by jwillikers — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Pull request merged**